versions/3.1.0.md

```diff
@@ -2336,6 +2336,18 @@ As such, inline schema definitions, which do not have a given id, *cannot* be us
 The [xml](#schemaXml) property allows extra definitions when translating the JSON definition to XML.
 The [XML Object](#xmlObject) contains additional information about the available options.
 
+###### Picking Schema Vocabularies
+
+OpenAPI documents are made up of multiple resources, sometimes resources are all in the same root OpenAPI document (inline resources), or resources are contain their own documents with `$ref` (separate resources). It is important for tooling to be able to detect what meta-schema any given resource wishes to be processed with: JSON Schema Core, JSON Schema Validation, OpenAPI Schema Object, or some custom meta schema.
```

Comment:
Can we reserve the use of the word "resource" to mean a separate file, HTTP-style resource or (:copyright: @handrews an octet-stream), rather than overloading it by using the term "inline resources". Are these better just called "components"?

I think we also need to address that not all resources are JSON Schema or even collections of JSON Schema. Resources can be pretty much arbitrary JSON/YAML as long as they comply with the object structure of the location they're being `$ref`d from.

`s/are contain/MAY be contained/`

I'm not clear on the use-case for setting the meta-schema to JSON Schema Core in an OAS document would be.

We should be consistent on whether it's `meta-schema` or `meta schema`.

Reply:
In JSON Schema world, we have started to refer to any schema which identifies itself as a "resource", which is probably where Phil is using the termonology from in this case. Specifically how schema resources can be embedded in other documents. If a JSON Schema has an `$id`, it's a "schema resource".

Reply:
I've changed this to be schema resource in a bunch of places. If the new version is no good lets start with new feedback and try to use the suggestion feature wherever possible to take a swing at making it better.